### DISCUSSION TITLE
HitDef Parameter Modifier

### DISCUSSION THREAD

**Discussion by SibilantScream** [ikemen-engine/Ikemen-GO#834] [Feature Request]
A general purpose hitdef parameter modifer that specifies which parameters to modify & whose hitdefs to apply changes to.

In state controller format this would look something like:

[HitDefModify]
FilterPlayer = p1,p2,enemy,ally,etc        ;player to apply to.
FilterAttrb = SCA,N S H T P
pausetime = 0                     ;Ignore pause & continute state
damage = damage * 0.5
palfx.time = 30
palfx.add = 255,255,255
(other HitDef parameters)

FilterAttrb in tandem with FilterPlayer should make possible reading on other players' hitdef since HitDefAttrb trigger can't do that.

In effect it's similar to KoF 2000's Armor Mode:
https://www.youtube.com/watch?v=5mKF6bcf5EI

**Comment by potsmugen** [comment]
This has existed for a while: ModifyHitdef.
